cis194/week5/stevemao/Var.hs

```diff
@@ -0,0 +1,35 @@
+{-# OPTIONS_GHC -Wall #-}
+{-# LANGUAGE FlexibleInstances #-}
+module Var where
+import qualified Data.Map as M
+import Homework
+
+class HasVars a where
+  var :: String -> a
+
+data VarExprT = Lit Integer
+           | Add VarExprT VarExprT
+           | Mul VarExprT VarExprT
+           | Var Integer
```

Comment:
To represent variable, it needs a name and we need to use `String` here at least to represent that.